nservicebus/messaging/moving-handlers.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Moving Handlers
+summary: Guidelines for choosing a way how ot move handlers between the endpoints
+reviewed: 2021-09-01
+component: Core
+isLearningPath: true
+---
+
+As the distributed systems evolve, sometimes there is a need to change which Endpoint handle which message type. During development phase such changes are relativelly straightforward, but when the system is in production additional consideration need to be given to the messages that are in-flight. 
+
+## How to move Handlers in between Endpoints
+
+To move a handler from SourceEndpoint to a DestinationEndpoint the following steps needs to be taken
+
+ - Copy the handler into DestinationEndpoint and deploy that endpoints. Now both endpoints are handling the same message type.
+ - Make sure that new Endpoint recieves the messages that it should be handling. From now on all newly send messages would only go to the DestinationEndpoint.
+ - After ensuring that the SourceEndpoint does not have any remaining messages of the type that should be processed by DestinationEndpoint, delete the handler from SourceEndpoint and deploy it.
+
```

Comment:
```suggestion
 - Copy the handler into `DestinationEndpoint` and deploy that endpoint. Now both endpoints are handling the same message type.
 - Make sure that `DestinationEndpoint` receives the messages that it should be handling. From now on all newly sent messages should only go to the DestinationEndpoint.
 - After ensuring that the `SourceEndpoint` does not have any remaining messages of the type that should be processed by `DestinationEndpoint`, delete the handler from `SourceEndpoint` and deploy it.

```

Reply:
This example assumes a single sending endpoint. Should there be some consideration around the behavior when there are multiple sending endpoints? In that case, you would update sender by sender, but that also means that the ´SourceEndpoint´ would still be receiving messages by senders that haven't been updated, right?

Reply:
Yeah, good catch. I will add that